explainer.md

```diff
@@ -6,58 +6,191 @@
 
 >A brief, 4-5 paragraph explanation of the feature’s value. Outline what the feature does and how it accomplishes those goals (in prose). If your feature creates UI, this is a great place to show mocks and user flows.
 
+With emerging AI innovations in both software and hardware ecosystem, one of the main challenges for the web is to bridge this software and hardware development and bring together a solution that scales across hardware platforms and works with any framework for web AI experiences. We propose the WebNN API as an abstraction for neural networks in the web browsers.
+
+![WebNN architecture](content/webnn_arch.png)
+
+As illustrated in the architecture diagram of the figure above, web browsers may implement the WebNN API using native machine learning API available in the operating system. This architecture allows Javascript frameworks to tap into cutting-edge machine learning innovations in the operating system and the hardware platform underneath it without being tied to platform-specific capabilities, bridging the gap between software and hardware through a hardware-agnostic abstraction layer. 
+
+At the heart of neural networks is a computational graph of mathematical operations. These operations are the building blocks of modern AI and machine learning technologies in computer vision, natural language processing, and robotics. 
+
+The WebNN API is a specification for constructing and executing computational graphs of neural networks. It provides web applications with the ability to create, compile, and run machine learning networks on the web browsers. The WebNN API may be implemented in web browsers using the available native operating system machine learning APIs for the best performance and reliability of results. The following code sample illustrates a simple usage of this API.
+
+``` Javascript
+const operandType = { type: 'float32', dimensions: [2, 2] };
+const context = navigator.ml.getNeuralNetworkContext().
+const builder = context.createModelBuilder();
+// 1. Create a model of the computational graph 'C = 0.2 * A + B'
+const constant = builder.constant(0.2);
+const A = builder.input('A', operandType);
+const B = builder.input('B', operandType);
+const C = builder.add(builder.mul(A, constant), B);
+const model = builder.createModel({'C': C});
+// 2. Compile the model into executable.
+const compilation = await model.compile();
+// 3. Bind inputs to the model and execute for the result
+const bufferA = new Float32Array(4).fill(1.0);
+const bufferB = new Float32Array(4).fill(0.8);
+let outputs = await compilation.compute({'A': { buffer: bufferA }, 'B': { buffer: bufferB }});
+// The computed result of [[1, 1], [1, 1]] is in the buffer associated with the output operand
+console.log(outputs.C.buffer);
```

Comment:
it would be good to output its shape, e.g.
```js
console.log(outputs.C.dimensions);
```

Reply:
Fixed.

Reply:
Thanks. 
I may not state clearly. I meant to add the output of shape besides the values, e.g.
```js
console.log("Output shape: " + outputs.C.dimensions);
console.log("Output value: " + outputs.C.buffer);
```

Reply:
Created https://github.com/webmachinelearning/webnn/pull/111 to follow up.